### PR TITLE
Sample Provider Create Requests

### PR DESCRIPTION
**Pull request by jntullo**
There have been many questions and some BZs related to the creation of new providers, so I have begun to add sample requests for them (listed in alphabetical order).

Keeping this WIP until I have a larger collection. Open to suggestions for where this would best be organized - should it be under "Provider Create" or its own section?

cc: @abellotti @mkoura

### CODE REVIEW

File: api/reference/providers.adoc [hunk 1]
```diff
@@ -452,3 +453,105 @@ NOTE: When managing providers of _provider_ provider_class, Tagging and Policy m
...
+            "type":"AuthToken",
+            "auth_key":"<auth_key_here>"
+         }
+      }
+   ]
+}
+----
+
+Openstack Container Manager:
+[source,json]
```
Comment (tbielawa) [inline api/reference/providers.adoc]: This is actually for the *OpenShift* container manager

**Comment by jntullo** [comment]
@miq-bot remove_label wip

File: api/reference/providers.adoc [hunk 2]
```diff
@@ -226,7 +227,7 @@ Compound credentials set:
 [[specifying-connection-configurations]]
 === Specifying Connection Configurations
 When creating or updating providers, connection configurations can be set. Connection configurations can be used
-to specify resources such as an amqp event provider for Openstack or for adding Hawkular metrics.
+to specify resources such as an amqp event provider for Openstack, adding Hawkular metrics, or an Authentication token.
 
 Hawkular metrics:
```
Comment (beeva-zeusarias) [inline api/reference/providers.adoc]: C&U metrics RHV:

                         "connection_configurations": [
                             {
                                 "endpoint": {
                                     "role": "default",
                                     "verify_ssl": 0
                                 }
                             },
                             {
                                 "endpoint": {
                                     "hostname": "host",
                                     "port": "port_database",
                                     "path": "ovirt_engine_history",
                                     "role": "metrics"
                                 }
                             }
                         ]
                     }

Comment (dkholodo) [inline api/reference/providers.adoc]: Is it possible an Openstack Cloud Manager also link to an existing Openstack Infra Provider?

File: api/reference/providers.adoc [hunk 3]
```diff
@@ -452,3 +453,247 @@ NOTE: When managing providers of _provider_ provider_class, Tagging and Policy m
...
+            "authtype":"bearer",
+            "type":"AuthToken",
+            "auth_key":"<auth key>"
+         }
+      }
+   ]
+}
+----
+
+Openstack Cloud Manager:
```
Comment (dkholodo) [inline api/reference/providers.adoc]: Is it possible an Openstack Cloud Manager also link to an existing Openstack Infra Provider?

**Comment by dayleparker** [comment]
Hi @jntullo @snecklifter - just checking if there's any remaining work to be done on this? (@dkholodo, have you gotten an answer elsewhere?) 
The content looks good to me, please let me know if I'm OK to merge it and cherry-pick to gaprindashvili.

File: api/reference/providers.adoc [hunk 4]
```diff
@@ -226,7 +227,7 @@ Compound credentials set:
 [[specifying-connection-configurations]]
 === Specifying Connection Configurations
 When creating or updating providers, connection configurations can be set. Connection configurations can be used
-to specify resources such as an amqp event provider for Openstack or for adding Hawkular metrics.
+to specify resources such as an amqp event provider for Openstack, adding Hawkular metrics, or an Authentication token.
```
Comment (dayleparker) [inline api/reference/providers.adoc]: Sorry for the nitpicks - can we make "authentication token" lower-case, and I would suggest marking up "ampq" with ` symbols so it doesn't get translated accidentally and stands out more. Thanks!

File: api/reference/providers.adoc [hunk 5]
```diff
@@ -452,3 +453,247 @@ NOTE: When managing providers of _provider_ provider_class, Tagging and Policy m
...
+      "authentication": {
+        "authtype": "bearer",
+        "auth_key": "<token_here>"
+      }
+    }
+  ]
+}
+----
+
+Microsoft System Center VMM:
```
Comment (dayleparker) [inline api/reference/providers.adoc]: Please add (SCVMM) in brackets after so it's easier for readers to ctrl+f.

File: api/reference/providers.adoc [hunk 6]
```diff
@@ -452,3 +453,247 @@ NOTE: When managing providers of _provider_ provider_class, Tagging and Policy m
...
+      "authentication": {
+        "username": "username",
+        "password": "password"
+      }
+    }
+  ]
+}
+----
+
+Openshift Container Manager:
```
Comment (dayleparker) [inline api/reference/providers.adoc]: Please capitalize the S --> OpenShift.

File: api/reference/providers.adoc (hunk 3 above)
Comment (dayleparker) [reply api/reference/providers.adoc]: One more nitpick - please capitalize the S in OpenStack.

File: api/reference/providers.adoc [hunk 7]
```diff
@@ -452,3 +453,247 @@ NOTE: When managing providers of _provider_ provider_class, Tagging and Policy m
...
+  "hostname": "sample_vmware.provider.com",
+  "ipaddress": "100.200.300.1",
+  "credentials": {
+    "userid": "uname",
+    "password": "pword"
+  }
+}
+----
+
+Vmware Cloud Manager:
```
Comment (dayleparker) [inline api/reference/providers.adoc]: Pleas capitalize VMware as such.

**Comment by dayleparker** [comment]
I added a few small comments, but looks ready to merge otherwise 👍

**Comment by cbudz** [comment]
Hi @jntullo, I'm going to be taking this over from Dayle at the moment. She left you some comments on the files regarding some small changes. If you can make those updates I'll go ahead and merge this PR. Please let me know if you have any questions.

Thanks,
Chris

**Comment by cbudz** [comment]
I've updated this doc based on the suggestions from @dayleparker. Merging and backporting to G and H branches.

**Pull request merged**